static/src/javascripts/projects/common/utils/detect.js

```diff
@@ -428,6 +437,26 @@ define([
         sacrificialAd.appendTo(document.body);
         return sacrificialAd;
     }
+    // end sync adblock detection
+
+    var getAdBlockers = new Promise(function (resolve) {
+        if (has(window.guardian.adBlockers, 'generic')) {
+            // adblock detection has completed
+            resolve(window.guardian.adBlockers);
+        } else {
+            // Push a listener for when the JS loads
+            window.guardian.adBlockers.onDetect = resolve.bind(this, window.guardian.adBlockers);
```

Comment:
I likez it :smile: You could use `null` instead of `this` to make sure the context is right. Not sure if that's the case, but is there a polyfill for `bind` already loaded at this stage? (mostly for IE8 users)